src/TorchSharp/TorchVision/dsets/MNISTReader.cs

```diff
@@ -0,0 +1,105 @@
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="path">Path to the folder containing the image files.</param>
+        /// <param name="prefix">The file name prefix, either 'train' or 't10k' (the latter being the test data set).</param>
+        /// <param name="device">The device, i.e. CPU or GPU to place the output tensors on.</param>
+        /// <param name="transform"></param>
+        public MNISTReader(string path, string prefix, Device device = null, ITransform transform = null)
+        {
+            // The MNIST data set is small enough to fit in memory, so let's load it there.
+
+            this.transform = transform;
+
+            var dataPath = Path.Combine(path, prefix + "-images-idx3-ubyte");
+            var labelPath = Path.Combine(path, prefix + "-labels-idx1-ubyte");
+
+            var count = -1;
+            var height = 0;
+            var width = 0;
+
+            byte[] dataBytes = null;
+            byte[] labelBytes = null;
+
+            using (var file = File.Open(dataPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            using (var rdr = new System.IO.BinaryReader(file)) {
+
+                var reader = new BigEndianReader(rdr);
+                var x = reader.ReadInt32(); // Magic number
+                count = reader.ReadInt32();
+
+                height = reader.ReadInt32();
+                width = reader.ReadInt32();
+
+                // Read all the data into memory.
+                dataBytes = reader.ReadBytes(height * width * count);
+            }
+
+            using (var file = File.Open(labelPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            using (var rdr = new System.IO.BinaryReader(file)) {
+
+                var reader = new BigEndianReader(rdr);
+                var x = reader.ReadInt32(); // Magic number
+                var lblcnt = reader.ReadInt32();
+
+                if (lblcnt != count) throw new InvalidDataException("Image data and label counts are different.");
+
+                // Read all the data into memory.
+                labelBytes = reader.ReadBytes(lblcnt);
+            }
+
+            var imgSize = height * width;
+
+            // Go through the data and create tensors
+            for (var i = 0; i < count; i++) {
+                var imgStart = i * imgSize;
+
+                data.Add(tensor(dataBytes[imgStart..(imgStart+imgSize)].Select(b => b/256.0f).ToArray(), new long[] { width, height }));
+                labels.Add(tensor(labelBytes[i], int64));
+            }
+        }
+
+        private ITransform transform;
+
+        public override long Count => data.Count;
+
+        private List<Tensor> data = new();
+        private List<Tensor> labels = new();
+
+        public override void Dispose()
+        {
+            data.ForEach(d => d.Dispose());
+            labels.ForEach(d => d.Dispose());
+        }
+
+        public override Dictionary<string, Tensor> GetTensor(long index)
```

Comment:
It's always helpful to copy the doc strings from the base class or interface where a method is introduced to all locations where it is overridden / implemented